src/login/logind-session.c

```diff
@@ -170,23 +170,16 @@ void session_set_user(Session *s, User *u) {
         user_update_last_session_timer(u);
 }
 
-int session_set_leader(Session *s, pid_t pid) {
-        _cleanup_(pidref_done) PidRef pidref = PIDREF_NULL;
+int session_set_leader(Session *s, PidRef leader) {
         int r;
 
         assert(s);
+        assert(pidref_is_set(&leader));
```

Comment:
assert is not right here, as it will be removed depending on the compiler options - instead just check and return an error

Reply:
Calling the function w/o leader set is programming error so isn't assert() better option here?

Reply:
I might be confused with the other calls, if this one doesn't have side effects and it's just a check then it's fine